include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -33,6 +38,88 @@ namespace execution
 inline namespace __dpl
 {
 
+namespace __internal
+{
+
+#if _ONEDPL_PREDEFINED_POLICIES
+struct __global_instance_tag {};
+#endif
+
+class alignas(sycl::queue) __queue_holder
+{
+    static_assert(sizeof(sycl::queue) >= sizeof(void*));
+    static_assert(alignof(sycl::queue) >= alignof(void*));
+
+    std::byte __buf[sizeof(sycl::queue)];
+
+    bool __has_queue() const
+    {
+        bool res = (nullptr != reinterpret_cast<void* const&>(*this));
```

Comment:
What about using something like
```C++
union Data
{
    void* raw;
    std::byte __buf[sizeof(sycl::queue)];
};
```
Does this data structure able to help us simplify your line above?